Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -1194,29 +1194,49 @@ class BrowserViewController: UIViewController {
                 break
             }
             tab.userScriptManager?.isU2FEnabled = webView.hasOnlySecureContent
-            if tab.contentIsSecure && !webView.hasOnlySecureContent {
-                tab.contentIsSecure = false
+            if tab.secureContentState == .secure && !webView.hasOnlySecureContent {
+                tab.secureContentState = .insecure
             }
             
-            topToolbar.contentIsSecure = tab.contentIsSecure
+            topToolbar.secureContentState = tab.secureContentState
         case .serverTrust:
             guard let tab = tabManager[webView] else {
                 break
             }
 
-            tab.contentIsSecure = false
-            topToolbar.contentIsSecure = tab.contentIsSecure
-
+            tab.secureContentState = .insecure
+            
             guard let serverTrust = tab.webView?.serverTrust else {
+                if let url = tab.webView?.url {
```

Comment:
This chain of `if` conditions is a bit complex, wdyt about refactoring it to be more readable?
Could be something like this, please recheck the logic
```
guard let serverTrust = tab.webView?.serverTrust else {
    if let url = tab.webView?.url {
        if url.isAboutHomeURL || url.isAboutURL {
            tab.secureContentState = .localHost
            topToolbar.secureContentState = .localHost
            break
        }

        if url.isReaderModeURL || url.isLocal {
            tab.secureContentState = .unknown
            topToolbar.secureContentState = .unknown
            break
        }
    }

    // fallback
    topToolbar.secureContentState = tab.secureContentState
    break
}
```